### PR TITLE
Scheduled spark app updates

### PR DESCRIPTION
**Pull request by deasydoesit**

### CODE REVIEW

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml [hunk 1]
```diff
@@ -16,74 +16,81 @@ data:
...
-      - key: dedicated
-        operator: Equal
-        value: spot-spark
-        effect: NoSchedule
-    nodeSelector:
-      nodegroup-type: spot-spark
-    labels:
-      version: 3.4.0
-    volumeMounts:
+  # Run 15 minutes after iot-reward-share-delta-lake-sink
```
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:24]: Running after the proper k8s cron?

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml [hunk 2]
```diff
@@ -16,74 +16,81 @@ data:
...
+  schedule: "25 */4 * * *"
```
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:25]: Running after the proper k8s cron?

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml [hunk 3]
```diff
@@ -16,74 +16,81 @@ data:
...
+  concurrencyPolicy: Forbid
+  successfulRunHistoryLimit: 1
+  failedRunHistoryLimit: 5
+  template:
+    type: Scala
+    mode: cluster
+    image: "public.ecr.aws/k0m1p4t7/spark:v3.4.0-aws"
+    imagePullPolicy: Always
+    mainClass: Main
+    mainApplicationFile: "s3a://foundation-data-lake-requester-pays/jars/spark-streaming-sql-assembly-1.0.2.jar"
```
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:35]: Update the jar version

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml [hunk 4]
```diff
@@ -16,74 +16,81 @@ data:
...
+      memory: "512m"
+      nodeSelector:
+        node.kubernetes.io/instance-type: m5.large
+      envVars:
+        TABLE_IOT_REWARD_SHARE: s3a://foundation-data-lake-requester-pays/bronze/iot_reward_share
+        PARTITION_BY: "date"
+        CHECKPOINT: s3a://foundation-data-lake-requester-pays/checkpoints/iot-hotspot-rewards
+        OUTPUT: s3a://foundation-data-lake-requester-pays/silver/iot-hotspot-rewards
+        QUERY_PATH: /app/query.sql
+        TRIGGER: AvailableNow
```
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:71]: Added to align with `iot-packets-silver` but not sure if needed

File: manifests/poc-data-cluster/prod/spark/iot-poc-silver.yaml
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/iot-poc-silver.yaml:1]: Updated to align with the others but we'll be commenting out on Friday

File: manifests/poc-data-cluster/prod/spark/mobile-hotspot-rewards-silver.yaml [hunk 5]
```diff
@@ -17,74 +17,79 @@ data:
...
-        operator: Equal
-        value: spot-spark
-        effect: NoSchedule
-    nodeSelector:
-      nodegroup-type: spot-spark
-    labels:
-      version: 3.4.0
-    volumeMounts:
+  # Run 15 minutes after mobile-reward-share-delta-lake-sink
+  schedule: "25 */4 * * *"
```
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/mobile-hotspot-rewards-silver.yaml:26]: Running after the proper k8s cron?

File: manifests/poc-data-cluster/prod/spark/recently-rewarded-hotspots-gold.yaml
Comment (deasydoesit) [inline manifests/poc-data-cluster/prod/spark/recently-rewarded-hotspots-gold.yaml:1]: This job was already a `ScheduledSparkApplication`. 

I updated the `spark-streaming-sql-assembly` and added the `TRIGGER` env var, but not sure if needed or not

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml (hunk 2 above)
Comment (ChewingGlass) [reply manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:25]: Yep

File: manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml (hunk 4 above)
Comment (ChewingGlass) [reply manifests/poc-data-cluster/prod/spark/iot-hotspot-rewards-silver.yaml:71]: Yeah that's the only change needed

File: manifests/poc-data-cluster/prod/spark/recently-rewarded-hotspots-gold.yaml
Comment (ChewingGlass) [reply manifests/poc-data-cluster/prod/spark/recently-rewarded-hotspots-gold.yaml:1]: This one shouldn't be needed since BATCH_PROCESSING is true

**Review by ChewingGlass — CHANGES_REQUESTED** [review]
Don't need trigger on the gold table since BATCH_PROCESSING is true

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**